apps/OboeTester/app/src/main/cpp/jni-bridge.cpp

```diff
@@ -1052,4 +1054,170 @@ Java_com_mobileer_oboetester_TestRapidCycleActivity_getCycleCount(JNIEnv *env, j
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_setCpuAffinityForCallback(JNIEnv *env, jobject thiz,
+                                                                                 jint mask) {
+    return AudioWorkloadTest::setCpuAffinityForCallback(mask);
+}
+
+JNIEXPORT jint JNICALL
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_getXRunCount(JNIEnv *env, jobject thiz) {
+    return sAudioWorkload.getXRunCount();
+}
+
+JNIEXPORT jint JNICALL
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_getCallbackCount(JNIEnv *env, jobject thiz) {
+    return sAudioWorkload.getCallbackCount();
+}
+
+JNIEXPORT jlong JNICALL
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_getLastDurationNs(JNIEnv *env, jobject thiz) {
+    return sAudioWorkload.getLastDurationNs();
+}
+
+JNIEXPORT jboolean JNICALL
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_isRunning(JNIEnv *env, jobject thiz) {
+    return sAudioWorkload.isRunning();
+}
+
+JNIEXPORT jint JNICALL
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_stop(JNIEnv *env, jobject thiz) {
+    return sAudioWorkload.stop();
+}
+
+JNIEXPORT jint JNICALL
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_close(JNIEnv *env, jobject thiz) {
+    return sAudioWorkload.close();
+}
+
+JNIEXPORT jobject JNICALL
+Java_com_mobileer_oboetester_AudioWorkloadTestActivity_getCallbackStatistics(JNIEnv *env, jobject obj) {
+    std::vector<AudioWorkloadTest::CallbackStatus> cppCallbackStats = sAudioWorkload.getCallbackStatistics();
+    jclass callbackStatusClass = env->FindClass("com/mobileer/oboetester/AudioWorkloadTestActivity$CallbackStatus");
+    if (callbackStatusClass == nullptr) {
+        std::cerr << "Error: Could not find Java class CallbackStatus" << std::endl;
+        return nullptr;
+    }
+
+    jmethodID callbackStatusConstructor = env->GetMethodID(callbackStatusClass, "<init>", "(IJJII)V");
+    if (callbackStatusConstructor == nullptr) {
+        std::cerr << "Error: Could not find constructor for CallbackStatus" << std::endl;
+        return nullptr;
+    }
+
+    jclass arrayListClass = env->FindClass("java/util/ArrayList");
+    if (arrayListClass == nullptr) {
+        std::cerr << "Error: Could not find Java class ArrayList" << std::endl;
+        return nullptr;
+    }
+
+    jmethodID arrayListConstructor = env->GetMethodID(arrayListClass, "<init>", "()V");
+    if (arrayListConstructor == nullptr) {
+        std::cerr << "Error: Could not find constructor for ArrayList" << std::endl;
+        return nullptr;
+    }
+
+    jmethodID arrayListAdd = env->GetMethodID(arrayListClass, "add", "(Ljava/lang/Object;)Z");
+    if (arrayListAdd == nullptr) {
+        std::cerr << "Error: Could not find 'add' method for ArrayList" << std::endl;
+        return nullptr;
+    }
+
+    jobject javaList = env->NewObject(arrayListClass, arrayListConstructor);
+
+    for (const auto& status : cppCallbackStats) {
+        jobject javaStatus = env->NewObject(
```

Comment:
Add comment that says this must match the constructor in  AudioWorkloadTestActivity.java